packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -69,6 +69,13 @@ export class DiscoveryWatcher {
           projectConfig,
           blockNumber,
         )
+
+        if (discovery.contracts.some((c) => c.errors !== undefined)) {
+          throw new Error(
+            `Errors occurred during discovery of ${projectConfig.name}`,
+          )
+        }
```

Comment:
Maybe it's better to retry this? Or set the retry policy of the taskQueue to never retry.